### PR TITLE
Remove unused Code S to Z

### PR DESCRIPTION
**Pull request by CasparKielwein**
Removes unused code in directories starting with Letters S to Z, as described in #1955 .

Pretty much analogue to #1952

### CODE REVIEW

File: code/ship/ship.h [hunk 1]
```diff
@@ -1545,11 +1540,9 @@ extern int ship_query_general_type(ship *shipp);
 extern int ship_docking_valid(int docker, int dockee);
 extern int get_quadrant(vec3d *hit_pnt, object *shipobjp = NULL);	//	Return quadrant num of given hit point.
 
-extern void ship_obj_list_rebuild();	// only called by save/restore code
 extern int ship_query_state(char *name);
 
 // Goober5000
```
Comment (Goober5000) [inline code/ship/ship.h]: Remove the `// Goober5000` comment too, because I didn't write the secondary function.

File: code/weapon/weapon.h [hunk 2]
```diff
@@ -476,7 +470,6 @@ typedef struct weapon_info {
     inline bool is_homing()        { return wi_flags[Weapon::Info_Flags::Homing_heat, Weapon::Info_Flags::Homing_aspect, Weapon::Info_Flags::Homing_javelin]; }
     inline bool is_locked_homing() { return wi_flags[Weapon::Info_Flags::Homing_aspect, Weapon::Info_Flags::Homing_javelin]; }
     inline bool hurts_big_ships()  { return wi_flags[Weapon::Info_Flags::Bomb, Weapon::Info_Flags::Beam, Weapon::Info_Flags::Huge, Weapon::Info_Flags::Big_only]; }
-    inline bool is_interceptable() { return wi_flags[Weapon::Info_Flags::Fighter_Interceptable, Weapon::Info_Flags::Turret_Interceptable]; }
```
Comment (Goober5000) [inline code/weapon/weapon.h]: Keep this function.

File: code/sound/audiostr.cpp [hunk 3]
```diff
@@ -1026,20 +1026,6 @@ void audiostream_set_volume_all(float volume, int type)
 	}
 }
 
-void audiostream_set_volume(int i, float volume)
```
Comment (asarium) [inline code/sound/audiostr.cpp]: This might still be useful so it should be kept.

Comment (CasparKielwein) [reply code/sound/audiostr.cpp]: ok

**Pull request merged**